test/runtests.py

```diff
@@ -571,10 +571,16 @@ def timeout_func(timeout_data):
         if not return_code_only:
             # check output
             if 'baseline' not in test:
-                # output lines must be 'pass' or 'passed' or empty
-                lines = (line.lower() for line in js_output.split(b'\n'))
-                if any(line != b'' and line != b'pass' and line != b'passed'
-                        for line in lines):
+                # output lines must be 'pass' or 'passed' or empty with at least 1 not empty
+                passes = 0
+                for line in js_output.split(b'\n'):
+                    if line !=b'':
+                        low = line.lower()
+                        if low == b'pass' or low == b'passed':
+                            passes = 1
+                        else:
+                            return self._show_failed(**fail_args)
+                if passes == 0:
```

Comment:
This change will cause missing test files OR tests that just don't run to count as fails rather than passes. Unfortunately this is macOS/linux only for now but figured that's better than nothing.